src/Command/ListDatabases.php

```diff
@@ -99,7 +99,6 @@ public function __construct(array $options = [])
      * Execute the operation.
      *
      * @see Executable::execute()
-     * @param Server $server
      * @return array An array of database info structures
```

Comment:
Any reason not to use an `array` return type hint on `execute()` here? Correct me if I'm wrong, but I think that'd be permitted by [covariance](https://www.php.net/manual/en/language.oop5.variance.php).

If this is out of scope for this PR, feel free to ignore.

Reply:
Covariance would indeed allow us to add an `array` type here. However, backward compatibility promise won't allow us to do so, as anyone extending `ListDatabases` would be encountering fatal errors. Granted, it doesn't make much sense extending these classes, but they are all marked as part of the API and non-final, so we can't add return types to any public or protected methods.

Reply:
Oh good point about extension. I didn't consider that.

Shame on us for not making these things `final` by default. Somewhere, @Ocramius is silently shaking his head in disappointment.